src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -184,15 +184,108 @@ private void MakeLoopAtomic()
                 case Notoneloop:
                     Type = Notoneloopatomic;
                     break;
-                case Setloop:
-                    Type = Setloopatomic;
-                    break;
                 default:
-                    Debug.Fail($"Unexpected type: {Type}");
+#if DEBUG
```

Comment:
Why is `#if DEBUG` required here? My understanding is that in release, `Debug.Assert` and everything that calls it is elided, and this includes any string interopolation being passed to it. Is that not the case?

Reply:
The TypeName property is under an `#if DEBUG`.  In a release build, it's as if here this was `Debug.Assert(condition. $"{SomePropertyThatDoesntActuallyExist}")`, which doesn't compile.

Reply:
ah yes